src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -585,6 +586,32 @@ internal sealed class RawArrayData
         public byte Data;
     }
 
+    // Subset of src\vm\methoddesc.hpp
+    [StructLayout(LayoutKind.Sequential)]
+    internal unsafe struct MethodDesc
+    {
+        public ushort Flags3AndTokenRemainder;
+        public byte ChunkIndex;
+        public byte Flags4; // Used to hold more flags
+        public ushort SlotNumber; // The slot number of this MethodDesc in the vtable array.
+        public ushort Flags; // See MethodDescFlags
+        public IntPtr CodeData;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private MethodDescChunk* GetMethodDescChunk() => (MethodDescChunk*)(((byte*)Unsafe.AsPointer<MethodDesc>(ref this)) - (sizeof(MethodDescChunk) + ChunkIndex * sizeof(IntPtr)));
+
+        public MethodTable* MethodTable => GetMethodDescChunk()->MethodTable;
+    }
+
+    internal unsafe struct MethodDescChunk
```

Comment:
Should this be LayoutKind.Sequential too?

Reply:
Probably. C# marks structs as LayoutKind.Sequential by default, but being explicit is the right thing to do.